src/Microsoft.ML.Core/Utilities/NormStr.cs

```diff
@@ -136,7 +136,7 @@ public NormStr Get(ReadOnlyMemory<char> str, bool add = false)
                 }
                 Contracts.Assert(ins == -1);
 
-                return add ? AddCore(str, hash) : null;
+                return add ? AddCore(str.ToString().AsMemory(), hash) : null;
```

Comment:
This is Proposal 3

Reply:
@sharwell and @justinormont proposed this. @justinormont also suggests adding a flag so that this conversion is optional, and that flag should be added to different methods up to FeaturizeText, where it should become an option for the user.

Reply:
`NormStr` being an internal class wouldn't expose the flag to users, only to the ML.NET code. Giving the ML.NET coders the choice to re-allocate the specific substring, or continue (potentially) saving memory and allocation time by using the original `NormStr`.

Reply:
I'm happy with not adding an option here, provided we do add a benchmark test.

Reply:
@sharwell's idea sounds the best to me. Please proceed with this and amend the PR to include only the proposal 3 changes.

Reply:
> NormStr being an internal class wouldn't expose the flag to users, only to the ML.NET code. Giving the ML.NET coders the choice to re-allocate the specific substring, or continue (potentially) saving memory and allocation time by using the original NormStr.

@justinormont My bad, I guess I misunderstood you. I thought you wanted to add this option all the way up to FeaturizeText, which is a public method, thus exposing the option to users. Since that wasn't what you suggested, then how up should this option go? I'd think it'd be okay for now to only create the flag `createNewStr` in the Get method, and default it to `true`, and only leave that option there, unless in the future we see the necessity to add this option to whatever methods call the `Get `method.